xblock/core.py

```diff
@@ -157,7 +157,13 @@ def open_local_resource(cls, uri):
         if "/." in uri:
             raise DisallowedFileError("Only safe file names are allowed: %r" % uri)
 
-        return pkg_resources.resource_stream(cls.__module__, os.path.join(cls.resources_dir, uri))
+        return cls.open_resource(uri)
+
+    @classmethod
+    def open_resource(cls, uri):
```

Comment:
```suggestion
    def _open_resource(cls, uri):
```

I imagine that you extracted this into its own function for mocking purposes, right? That is all good, but by defining this method on a public class, we would be adding a new public method to the XBlock API.

Public API method exert a maintenance burden, because we must support them indefinitely. They can only be removed by announcing a DEPR, waiting for it to be accepted, and releasing a new major XBlock version. So, I would not add a new public method unless there is a strong reason to do so.

Instead, we could avoid that by marking the method as "private" with a leading underscore. That way, we could change or remove it in the future without breaking the public contract. How does that sound?

Reply:
That sounds great. 

I learned not to introduce public things into libraries unless absolutely necessary.